cmd/bosun/expr/annotate.go

```diff
@@ -0,0 +1,254 @@
+package expr
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"math"
+
+	"bosun.org/cmd/bosun/expr/parse"
+	"bosun.org/models"
+	"bosun.org/opentsdb"
+	"github.com/MiniProfiler/go/miniprofiler"
+	"github.com/bosun-monitor/annotate"
+	"github.com/kylebrandt/boolq"
+)
+
+var Annotate = map[string]parse.Func{
+	// Funcs for querying elastic
+	"ancounts": {
+		Args:   []models.FuncType{models.TypeString, models.TypeString, models.TypeString},
+		Return: models.TypeSeriesSet,
+		Tags:   tagFirst,
+		F:      AnCounts,
+	},
+	"andurations": {
+		Args:   []models.FuncType{models.TypeString, models.TypeString, models.TypeString},
+		Return: models.TypeSeriesSet,
+		Tags:   tagFirst,
+		F:      AnDurations,
+	},
+	"antable": {
+		Args:   []models.FuncType{models.TypeString, models.TypeString, models.TypeString, models.TypeString},
+		Return: models.TypeTable,
+		F:      AnTable,
+	},
+}
+
+func procDuration(e *State, startDuration, endDuration string) (*time.Time, *time.Time, error) {
```

Comment:
also, why does this need to return pointers? If it is either an error, or valid times, just return `time.Time{}` in the error case. Saves some of our `*` budget.